### PR TITLE
DBZ-318 Support for a Decimale with variable scale schema

### PR DESCRIPTION
**Pull request by jpechane**

### CODE REVIEW

File: debezium-core/src/main/java/io/debezium/data/VariableScaleDecimal.java [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+    }
+
+    /**
+     * Decodes the encoded value - see {@link #fromLogical(Schema, BigDecimal)} for encoding format
+     * 
+     * @param value the encoded value
+     * @return the decoded value
+     */
+    public static BigDecimal toLogical(final Struct value) {
+        return new BigDecimal(new BigInteger((byte[])value.getBytes(VALUE_FIELD)), value.getInt16(SCALE_FIELD));
```
Comment (gunnarmorling) [inline debezium-core/src/main/java/io/debezium/data/VariableScaleDecimal.java]: s/getInt16/getInt32/.

That said the method seems not used at all?

File: debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/PostgresValueConverter.java [hunk 2]
```diff
@@ -96,14 +106,7 @@ public SchemaBuilder schemaBuilder(Column column) {
             case PgOid.TEXT_ARRAY:
                 return SchemaBuilder.array(SchemaBuilder.OPTIONAL_STRING_SCHEMA);
             case PgOid.NUMERIC_ARRAY:
-                switch (decimalMode) {
-                    case DOUBLE:
-                        return SchemaBuilder.array(Schema.OPTIONAL_FLOAT64_SCHEMA);
-                    case PRECISE:
-                        // values are fixed-precision decimal values with exact precision.
-                        // Use Kafka Connect's arbitrary precision decimal type and use the column's specified scale ...
-                        return SchemaBuilder.array(Decimal.builder(column.scale()).optional().build());
```
[the remark this answers is not in the document]
Comment (jpechane) [reply debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/PostgresValueConverter.java]: This is an oversight, thanks for spotting!

File: debezium-core/src/main/java/io/debezium/data/VariableScaleDecimal.java (hunk 1 above)
Comment (jpechane) [reply debezium-core/src/main/java/io/debezium/data/VariableScaleDecimal.java]: This is intentional - a java consumer might use this method to easilly process the datatype

Comment (gunnarmorling) [reply debezium-core/src/main/java/io/debezium/data/VariableScaleDecimal.java]: Ok, I don't mind having that method, but it must use `getInt32()` (we should have a quick test for it, too).

Comment (jpechane) [reply debezium-core/src/main/java/io/debezium/data/VariableScaleDecimal.java]: Fixed and a test added

**Comment by gunnarmorling** [comment]
Rebased, squashed and merged. Thanks, @jpechane!

**Pull request closed without merging**